pkg/mirrorregistries/generator_test.go

```diff
@@ -45,19 +48,90 @@ var _ = Describe("MirrorRegistriesConfig", func() {
 `
 	)
 
-	It("extract data from registries config", func() {
-		dataList, err := extractLocationMirrorDataFromRegistries(expectedFormatOutput)
+	It("extract data from registries config with mirrors", func() {
+		dataList, err := extractLocationMirrorDataFromRegistries(configWithMirrors)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(dataList).Should(Equal(expectedExtractList))
 	})
 
-	It("test get CA contents", func() {
-		file, err := ioutil.TempFile("", "ca.crt")
+	It("extract data from registries config without mirrors", func() {
+		_, err := extractLocationMirrorDataFromRegistries(configWithoutMirrors)
+		Expect(err.Error()).Should(Equal("failed to cast registry key to toml Tree, registriesConfToml: unqualified-search-registries = [\"registry1\", \"registry2\", \"registry3\"]"))
+	})
+
+	It("extract data from file with garbage", func() {
+		_, err := extractLocationMirrorDataFromRegistries(configWithGarbage)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("extract data from empty file", func() {
+		_, err := extractLocationMirrorDataFromRegistries("")
+		Expect(err.Error()).Should(Equal("failed to cast registry key to toml Tree, registriesConfToml: "))
+	})
+
+	It("check if mirrors configured - CA and registry.conf don't exist", func() {
+		m := mirrorRegistriesConfigBuilder{
+			MirrorRegistriesConfigPath:      "/tmp/this-file-for-sure-does-not-exist",
+			MirrorRegistriesCertificatePath: "/tmp/this-file-for-sure-does-not-exist",
+		}
+		res := m.IsMirrorRegistriesConfigured()
+		Expect(res).Should(Equal(false))
+	})
+
+	It("check if mirrors configured - registry.conf exists without registries, CA exists", func() {
+		tempRegistriesFile, err := ioutil.TempFile(os.TempDir(), "registries.*.conf")
+		defer os.Remove(tempRegistriesFile.Name())
 		Expect(err).NotTo(HaveOccurred())
-		_, err = file.WriteString("some ca data")
+		_, _ = tempRegistriesFile.WriteString(configWithoutMirrors)
+
+		tempCAFile, err := ioutil.TempFile(os.TempDir(), "ca.*.crt")
+		defer os.Remove(tempCAFile.Name())
 		Expect(err).NotTo(HaveOccurred())
-		contents, err := readFile(file.Name())
+		_, _ = tempCAFile.WriteString("some random CA certificate")
+
+		m := mirrorRegistriesConfigBuilder{
+			MirrorRegistriesConfigPath:      tempRegistriesFile.Name(),
+			MirrorRegistriesCertificatePath: tempCAFile.Name(),
+		}
+		res := m.IsMirrorRegistriesConfigured()
+		Expect(res).Should(Equal(false))
+	})
+
+	It("check if mirrors configured - registry.conf exists with garbage, CA exists", func() {
+		tempRegistriesFile, err := ioutil.TempFile(os.TempDir(), "registries.*.conf")
+		defer os.Remove(tempRegistriesFile.Name())
+		Expect(err).NotTo(HaveOccurred())
+		_, _ = tempRegistriesFile.WriteString(configWithGarbage)
+
+		tempCAFile, err := ioutil.TempFile(os.TempDir(), "ca.*.crt")
+		defer os.Remove(tempCAFile.Name())
+		Expect(err).NotTo(HaveOccurred())
+		_, _ = tempCAFile.WriteString("some random CA certificate")
```

Comment:
You could extract most of this into a `BeforeEach` and `AfterEach` in a context like `Context("with temp files"...` or something.

Then each test only needs to write the content to the file (which is the important part).

Reply:
Yes that makes sense